Bootstrap without inline code

dgrammatiko opened the pull request:
Pull Request for Issue # .
### Summary of Changes
- No more inline scipts for bootstrap components
- Make Joomla's bootstrap (js) dependancy less strict, as the init script together with the jlayouts dictate the js framework. For example someone can override bootstrap.js, bootstrap-init.js and the layouts and use Foundation or Ukit or any other framework!
### Testing Instructions
#### You will Need to apply https://github.com/joomla/joomla-cms/pull/11671 before testing!!!

For modals, tooltips, popover and tabs apply this patch and navigate to isis template on the backend. Ensure that no javascript errors are logged and everything still works.
To test the other bootstrap components download this module https://github.com/dgt41/bootstrap-test/raw/gh-pages/bootstrap_test.zip install, publish in the front end and then visit that page (template must be protostar).
Enjoy!
### B/C

There are few layouts that are touched, but joomla's b/c promise states that these are not B/C breaks. Down to PLT for a decision on that.
### Documentation Changes Required

None.

Calling @andrepereiradasilva @Fedik to review
@laoneo this might help your quest for bootstrap 3

okonomiyaki3000 committed:
Bugfixes and improvements for html5fallback. Fixes #6409

n9iels committed:
Move component message to system-message-container

n9iels committed:
Revert pkg_weblinks.xml

n9iels committed:
Move messages to view.html.php and revert changes is com_cache

n9iels committed:
Merge branch 'staging' of https://github.com/joomla/joomla-cms into component-messages

n9iels committed:
CS

n9iels committed:
CS + generate css from less

okonomiyaki3000 committed:
Add a 'check all' box for media lists. Fixes #8852

n9iels committed:
solve merge conflict

Hackwar committed:
New URL Routing!

Fixes #8256

wilsonge committed:
Merge branch 'staging' into 3.6.x

Hackwar committed:
Add missing files from the routing PR

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Merge branch 'form' into 3.6.x

wilsonge committed:
Merge branch 'form' into 3.6.x

wilsonge committed:
Merge branch 'form' into 3.6.x

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Merge branch 'nesteddelete' into 3.6.x

wilsonge committed:
Merge branch 'message' into 3.6.x

# Conflicts:
#	administrator/components/com_installer/views/install/tmpl/default.php
#	administrator/components/com_installer/views/update/tmpl/default.php
#	administrator/templates/isis/less/template.less

wilsonge committed:
Merge branch 'category' into 3.6.x

wilsonge committed:
Merge branch 'category' into 3.6.x

wilsonge committed:
Merge branch 'category' into 3.6.x

wilsonge committed:
Merge branch 'whereQueries' into 3.6.x

Fixes #8718

wilsonge committed:
Merge branch 'categories' into 3.6.x

Fixes #8634

Commit:
Version bump dev branch

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Merge branch 'staging' into 3.6.x

andrepereiradasilva committed:
site system to html5

izharaazmi committed:
Remove usage of deprecated JArrayHelper - com_config

izharaazmi committed:
Remove usage of deprecated JArrayHelper - com_menus

izharaazmi committed:
(2) Remove usage of deprecated JArrayHelper - com_menus

Commit:
JModelForm - add parameter group to preprocessData

Commit:
Change how JPluginHelper::load() caches data

mbabker committed:
Allow a custom page title to be defined for an article without a menu item. Fixes #10838

mbabker committed:
Templates are in the scope of JDocument. Fixes #10847

andrepereiradasilva committed:
use $this instead of $doc

nielsnuebel committed:
Merge pull request #4 from joomla/staging

Update to new Version

nielsnuebel committed:
Add sortable function to repeatable field

nielsnuebel committed:
Add JFactory::getDocument

nielsnuebel committed:
Add Space

nielsnuebel committed:
Add Jquery Document Ready

zero-24 committed:
JArrayHelper => ArrayHelper for the backend

zero-24 committed:
some missing

Commit:
Quote fields

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Bump version to 3.7-dev

Commit:
Convert to a lambda function and make the callback cache controller work with Closures

laoneo committed:
Move pagination list rendering to layout. Fixes #10982

dgrammatiko committed:
URL field using JLayout. Fixes #10995

dgrammatiko committed:
Text field using JLayouts. Fixes #11000

dgrammatiko committed:
Textarea field using JLayout. Fixes #11001

dgrammatiko committed:
Range Field using JLayout. Fixes #11002

dgrammatiko committed:
Password field using JLayout. Fixes #11003

dgrammatiko committed:
Number field using JLayout. Fixes #11006

dgrammatiko committed:
Meter field using JLayouts. Fixes #11007

dgrammatiko committed:
Hidden form field using JLayout. Fixes #11008

dgrammatiko committed:
Email form field using JLayout. Fixes #11009

Commit:
Make JAuthentication autoloaded, split 2 classes to 2 files. Fixes #11023

dgrammatiko committed:
Color Form Field using JLayouts. Fixes #11030

zero-24 committed:
merge conflichts

Commit:
Deprecate overriding pagination functions. Fixes #11247

Commit:
Use layouts in the vote plugin. Closes #11300

wilsonge committed:
Merge branch 'staging' into 3.6.x

wilsonge committed:
Merge branch 'staging' into 3.7.x

wilsonge committed:
Merge branch 'staging' into 3.7.x

wilsonge committed:
Merge branch 'array_helper' into 3.7.x

wilsonge committed:
Merge branch 'staging' into 3.7.x

wilsonge committed:
Merge branch 'array_helper' into 3.7.x. Fixes #10675

wilsonge committed:
Merge branch 'array_helper' into 3.7.x

infograf768 committed:
Implementing Menu Items modals (New feature)

Commit:
Add a plugin with a nag message to update PHP. Fixes #11498

wilsonge committed:
Merge branch 'staging' into 3.7.x

n9iels committed:
Add JHtmlBootstrap for front-end editing. Fixes #11562

coolcat-creations committed:
added show/hide introtext parameter to mod_articles_news

coolcat-creations committed:
added <?php if (!$params->get('intro_only')) : ?> back again since it might be a B/C setting/parameter

coolcat-creations committed:
added tabs in code

zero-24 committed:
Update mod_articles_news.xml

zero-24 committed:
Update _item.php

coolcat-creations committed:
Merge pull request #4 from zero-24/patch-14

default needs to be enabled by default for B/C

coolcat-creations committed:
corrected typo from last merge

coolcat-creations committed:
changed label and description

wilsonge committed:
Merge branch 'staging' into 3.7.x

wilsonge committed:
Merge branch 'repeatable' into 3.7.x

dgrammatiko committed:
No more inline scipts for bootstrap components

yvesh commented on media/system/js/bootstrap-init.js:
```diff
@@ -0,0 +1,245 @@
+!(function() {
```
@dgt41 Your are missing the declaration of jQuery here (else $ could still be mootools, you know the fun)..

```javascript
(function ($) {
// ...
}(jQuery));
```

For testability it would be nice if that whole code gets refactored into functions, but this is maybe something for a different PR..

Thank you for your work! Template team is very grateful!

dgrammatiko committed:
remove some extra lines

dgt41 commented on media/system/js/bootstrap-init.js:
```diff
@@ -0,0 +1,224 @@
+!(function() {
+	jQuery(document).ready(function($) {
```
@yvesh this line should take care of that! The first auto executing function is just an initiator and will not assign $ to jQuery, that is will happen in this line, so I think is ok

yvesh replied:
Oh overlooked that $ in the function, you are right. 👍

yvesh commented:
@dgt41 looks like you also need to update the php unit tests accordingly.. Travis is complaining

dgt41 commented:
@yvesh yes that's the fun part 🙊

dgrammatiko committed:
js strict mode, unit tests

dgrammatiko committed:
CS

dgrammatiko committed:
irrelevant code

dgt41 commented on libraries/cms/toolbar/button/popup.php:
```diff
@@ -85,9 +85,10 @@ public function fetchButton($type = 'Modal', $name = '', $text = '', $url = '',
+			JFactory::getDocument()->addScriptDeclaration('
+		jQuery("#modal-' . $name . '").on("hide", function () {' . $onClose . ';});
+			'
+			);
```
This script needs to be overridable, probably using JLayout

dgrammatiko committed:
allow better overriding of the scripts

dgrammatiko committed:
toolbar popup onclose overriding layout

dgrammatiko committed:
oops

C-Lodder commented:
@dgt41 - Could you define `$(this)` as a variable if it's used more than once please? Example:

**Before:**
```
var tooltipsSelector = $(this).data('selector'),
	animation = $(this).data('animation'),
	html  = $(this).data('html'),
	placement = $(this).data('placement'),
	title = $(this).data('title'),
	selector2 = $(this).data('other-selector'),
	trigger = $(this).data('trigger'),
	delay = $(this).data('delay'),
	container = $(this).data('container'),
	template = $(this).data('template'),
	$onshow = $(this).data('on-show'),
	$onshown = $(this).data('on-shown'),
	$onhide = $(this).data('on-hide'),
	$onhidden = $(this).data('on-hidden');
```
**After:**
```
var $self = $(this),
	tooltipsSelector = $self.data('selector'),
	animation = $self.data('animation'),
	html = $self.data('html'),
	placement = $self.data('placement'),
	title = $self.data('title'),
	selector2 = $self.data('other-selector'),
	trigger = $self.data('trigger'),
	delay = $self.data('delay'),
	container = $self.data('container'),
	template = $self.data('template'),
	$onshow = $self.data('on-show'),
	$onshown = $self.data('on-shown'),
	$onhide = $self.data('on-hide'),
	$onhidden = $self.data('on-hidden');
```

Sorry to be nit-picky, but a performance decrease springs to mind when I see creating the same jQuery object loads of times

dgrammatiko committed:
optimizations

C-Lodder commented:
@dgt41 - You've made some mistakes with `$self`. There will be some instances where you need to redefine it as it's out of scope:

```
var $self = $(this);
$('.modalTooltip').each(function(){
	var attr = $self.attr('data-placement');
});
```
As you can see from the above code, `attr` will not represent the `data-placement` attribute value for `$('.modalTooltip')` because you've defined `$self` outside `each(function() { ...`

If you want me to re-do it for you and put it on PasteBin, let me know and I'll send it over

dgrammatiko committed:
oops

dgt41 commented:
@C-Lodder thanks, I didn't pay much attention I guess

mbabker commented on layouts/libraries/cms/html/bootstrap/framework.php:
```diff
@@ -0,0 +1,17 @@
+$debug = empty($displayData['debug']) ? false : $displayData['debug'];
+
+// Load jQuery
+JHtml::_('jquery.framework');
```
CAN WE STOP MAKING LAYOUTS THAT ARE 100% JAVASCRIPT OR PHP FUNCTION CALLS!?

andrepereiradasilva replied:
ehehhe

mbabker commented:
This is exactly why JHtml has a registered callback system.

If you're hell bound and determined to move everything to layouts then `JHtml::register()`, `JHtml::unregister()`, `JHtml::isRegistered()`, `JHtml::_()`, `JHtml::call()`, and `JHtml::extract()` should be deprecated and everything refactored to just call the methods directly.

Until that happens, I honestly just see this PR as adding more complexity to an already complex and poorly constructed system in general.  We've already merged a ton of crap code and even crappier layout structures, can we stop doing this?  There is no freakin' reason at all to have a layout that has nothing but a closing div or nothing but pure PHP function calls.

dgrammatiko committed:
revert the script layout part

dgt41 commented:
@mbabker I reverted that part

Fedik commented:
@dgt41 you know I am against such stuff, against use an empty element to keep the options :neckbeard: 
it have no profit, really

it force +1 DOM search for each feature, and work with DOM *always* slower.
Here is huge drawback if compare to inline script: All code `bootstrap-init.js` will be *always* executed. Example DOM query `$('.joomla-accordion')` will be always run, even if `accordion` not exist on the page, and there more such queries in the file.

And `data-blabla=""` should be assigned to the element where it used (instead of empty random element), otherwise use scriptOptions.

dgt41 commented:
@Fedik Flexibility comes at a cost. This PR allows overriding of some hidden in the bootstrap.php code. Could it be done otherwise? Yes! In my code I override the bootstrap.php altogether. Maybe this could be broken to individual scripts per component as @yvesh suggested! Then we don't have useless empty elements. About the span/data: in some cases is really hard to not go this way. The idea here was to give some flexibility to template creators to move from 2.3.2 to 3 with minimum effort...

Fedik commented:
> This PR allows overriding of some hidden in the bootstrap.php code

yes I understood it, but the cost is to height, for my opinion 😉 
There should be a better way, I try to show you some example, in near future, I hope 😄 

I tried similar in #6357,  there "bootstrap feature" runs only if it requested

dgt41 commented:
@Fedik I forgot about #6357, yes that's a better BUT requires the event system which is not available (and probably won't be for 3.x)

Fedik committed:
Allow adding script options without inline JavaScript. Fixes #11671

wilsonge committed:
Merge #10795 into 3.7.x. Fixes #10795

Fedik commented:
@dgt41 check #11671, 
can take  #6357 without `Joomla.Behavior` part, 
mean only `behavior-xxxx.js` files and changes in `/html/`  and do something like:

```javascript
var accordion = Joomla.getOptions('bootstrap.accordion');
if(accordion){
  // Set up Bootstrap Accordion feature
}
```

In result we do not have useless DOM query, and do not have inline JavaScript 😉

mbabker committed:
New method for retrieving the strings from the store. Fixes #11692

dgrammatiko committed:
no crappy spans

dgt41 commented:
@Fedik @andrepereiradasilva please check

dgrammatiko committed:
UT

n9iels commented:
If I add option to the carousel, like this `JHtml::_('bootstrap.carousel', 'myCarousel', array("interval" => 8000));` nothing change...

Before I changed the interval to 8000 the slider slides approximately every 5 second. After I change the interval to 8000 this is still the same.

mbabker committed:
Allow registering callback functions for the debug console. Fixes #11700

dgt41 commented:
@n9iels I think the problem is that the number is parsed as text, I will review the code tomorrow. Thanks

infograf768 committed:
Using modal_category field in various core locations

dgrammatiko committed:
selectors mixed

dgt41 commented:
@n9iels `myCarousel` is an ID or a class? you have to specified that

n9iels commented:
I did not specified with a # or something like that. The case I described was based on you example module.

dgrammatiko committed:
UT

dgrammatiko committed:
UT

dgrammatiko committed:
clear the selectors mess

wilsonge committed:
Merge branch 'staging' into 3.7.x

wilsonge committed:
Merge branch 'staging' into 3.7.x

Commit:
Wrong class for chosen (#11820)

wilsonge committed:
Merge branch 'staging' into 3.7.x

infograf768 committed:
3.7.0: Using the new modal_menu field in various core places (#11694)

* Using the new modal_menu field in various core places

* Simplifying clearField

* Simplify conditional

* Parenthesis replaced by mdash

wilsonge committed:
Merge branch 'modal_category_field' into 3.7.x. Fixes #11717

wilsonge committed:
Merge branch '3.7.x' of github.com:joomla/joomla-cms into 3.7.x

infograf768 committed:
3.7.0: New Feature=>editor-xtd to insert menu items links in content (#11721)

* 3.7.0: New Feature=>editor-xtd to insert menu items links in content

* correcting some errors

* quoting for sql azure

* one more

* Changed icon + lang constants

wilsonge committed:
Merge branch 'staging' into 3.7.x

wilsonge committed:
Chosen upgrade. Fixes #11779

Squashed commit of the following:

commit 48051c7bf2792fb64cbe10046ada196fc9d233b8
Author: Brian Teeman <brian@teeman.net>
Date:   Thu Aug 25 14:31:04 2016 +0100

    css change only

commit 90a89fddf8e70d036b3779022a1477dc632f78ba
Author: Brian Teeman <brian@teeman.net>
Date:   Thu Aug 25 14:28:58 2016 +0100

    Revert "css fix thanks @C-Lodder"

    This reverts commit f7320cfb331e018b179c9356b642dac330f55c22.

commit f7320cfb331e018b179c9356b642dac330f55c22
Author: Brian Teeman <brian@teeman.net>
Date:   Thu Aug 25 14:09:03 2016 +0100

    css fix thanks @C-Lodder

commit 42c9de38e90c6a1ad48def219b65a0cc77159933
Author: Brian Teeman <brian@teeman.net>
Date:   Wed Aug 24 18:10:33 2016 +0100

    fix permissions

commit b8fdd34d3e5678bbe8cd568e1b32ec951ac18421
Author: Brian Teeman <brian@teeman.net>
Date:   Wed Aug 24 18:07:53 2016 +0100

    add minified version of js

commit 0ef056d06609cab17b272aa7c37da4b0521199b6
Author: Brian Teeman <brian@teeman.net>
Date:   Wed Aug 24 17:59:55 2016 +0100

    add documented js hacks

commit c68c275bfa9f87f287ac394baa7de65d32565c14
Author: Brian Teeman <brian@teeman.net>
Date:   Wed Aug 24 17:48:01 2016 +0100

    add comment block

commit 983dbf5a561b25aeef9f8d2658edc58e352da780
Author: Brian Teeman <brian@teeman.net>
Date:   Wed Aug 24 17:08:47 2016 +0100

    rename liszt: chosen:

commit 8caad7361bdddc25ae81912892061dd1b04f759f
Author: Brian Teeman <brian@teeman.net>
Date:   Wed Aug 24 17:07:47 2016 +0100

    Revert "rename liszt: chosen: and add JUI hacks"

    This reverts commit 86116716d6ff7ba307621de3b0e1fdc8f90bc2b1.

commit 86116716d6ff7ba307621de3b0e1fdc8f90bc2b1
Author: Brian Teeman <brian@teeman.net>
Date:   Wed Aug 24 17:06:21 2016 +0100

    rename liszt: chosen: and add JUI hacks

commit 535bea0c37a7bb6667a09fe39e3e99153e3b0bad
Author: Brian Teeman <brian@teeman.net>
Date:   Wed Aug 24 16:45:14 2016 +0100

    replace chosen: with liszt:

commit 2686df5323907e9a394a2a8279a45667ecda89d3
Author: Brian Teeman <brian@teeman.net>
Date:   Wed Aug 24 16:43:35 2016 +0100

    replace _chosen with _chzn

commit 691be65de111d1e6fb25e06a6037981ec13514bd
Author: Brian Teeman <brian@teeman.net>
Date:   Wed Aug 24 16:42:49 2016 +0100

    replace chosen- with chzn-

commit 863b5f9ec87e4702e6c70b5d9c8d705608cb2e5c
Author: Brian Teeman <brian@teeman.net>
Date:   Wed Aug 24 16:37:20 2016 +0100

    update js with original version

commit 537735c2a46513daac39fed29fe32c65e415c859
Author: Brian Teeman <brian@teeman.net>
Date:   Wed Aug 24 16:36:19 2016 +0100

    remove min js

commit e7395625704664bea75cd342757480ba7b6a7b62
Author: Brian Teeman <brian@teeman.net>
Date:   Wed Aug 24 16:35:21 2016 +0100

    replace chosen- with chzn-

commit db8243160065d8f26b606bcfb436dad8e8db929f
Author: Brian Teeman <brian@teeman.net>
Date:   Wed Aug 24 16:33:46 2016 +0100

    update css file with original version

commit 24a955be243ff8a42da38b1c98a07149e7cd86c0
Author: Brian Teeman <brian@teeman.net>
Date:   Wed Aug 24 16:29:34 2016 +0100

    update sprites

commit 4856d5e0be4ad0152dda452b6d0a5669fac1c4b1
Merge: 4ddd833 f10ab35
Author: Brian Teeman <brian@teeman.net>
Date:   Tue Aug 23 16:23:34 2016 +0100

    Merge branch 'staging' of https://github.com/joomla/joomla-cms into staging

commit 4ddd833e3d370ae9accc2ddb57a16e4302eb5981
Merge: b5ffa27 8258906
Author: Brian Teeman <brian@teeman.net>
Date:   Mon Aug 22 10:30:58 2016 +0100

    Merge branch 'staging' of https://github.com/joomla/joomla-cms into staging

commit b5ffa2745a6e8f32e87963dbe234f25b7882ace5
Merge: 24e6c45 e80e319
Author: Brian Teeman <brian@teeman.net>
Date:   Sun Aug 21 17:45:45 2016 +0100

    Merge branch 'staging' of https://github.com/joomla/joomla-cms into staging

commit 24e6c45f87dd15bb58d4968df31024f1b79dd01f
Merge: 962b424 c1b0295
Author: Brian Teeman <brian@teeman.net>
Date:   Fri Aug 19 21:08:30 2016 +0100

    Merge branch 'staging' of https://github.com/joomla/joomla-cms into staging

commit 962b424dc0edaed79581179c769951e4944ff465
Merge: 14f5b1c 4c539c4
Author: Brian Teeman <brian@teeman.net>
Date:   Thu Aug 18 12:06:49 2016 +0100

    Merge branch 'staging' of https://github.com/joomla/joomla-cms into staging

commit 14f5b1caf6a37ad396fab2c860450688f76142a2
Merge: 2baa00b edb70c2
Author: Brian Teeman <brian@teeman.net>
Date:   Mon Aug 15 10:06:07 2016 +0100

    Merge branch 'staging' of https://github.com/joomla/joomla-cms into staging

commit 2baa00b27295ad037e22685e9e3c73097e9ce5d7
Merge: 97d1fb7 f827dfc
Author: Brian Teeman <brian@teeman.net>
Date:   Sun Aug 14 22:38:02 2016 +0100

    Merge branch 'staging' of https://github.com/joomla/joomla-cms into staging

commit 97d1fb7fce493acbfccaf1ffde30cd05a5d3f04a
Author: Brian Teeman <brian@teeman.net>
Date:   Sun Aug 14 16:28:13 2016 +0100

    Revert "toolbar"

    This reverts commit 8f995f199efdfe8668c5f6d7f28a733076f89383.

commit 8f995f199efdfe8668c5f6d7f28a733076f89383
Author: Brian Teeman <brian@teeman.net>
Date:   Sun Aug 14 16:27:57 2016 +0100

    toolbar

mbabker committed:
Reduce use of JFactory::getConfig() where practical

bertmert committed:
Two search modules on same page: Invalid HTML. (#11229)

* Add postfix for unique id ($module->id)

* Respect postfixes for unique id in mod_search

wilsonge committed:
Merge branch 'staging' into 3.7.x

vinespie committed:
Adding "category on fly" functionality when create/edit contact. Fixes #11354

wilsonge committed:
Merge branch 'introtext_mod_news' into 3.7.x. Fixes #11564

andrepereiradasilva committed:
[html5] Hathor administrator template. Fixes #10634

wilsonge committed:
Merge branch 'system_frontend_html5' into 3.7.x. Fixes #10641

Commit:
[Optimize] Generate thumbnail image method to avoid pixelated error in thumbnails. Fixes #7919

wilsonge committed:
Fix since tag

brianteeman committed:
Normalise Block this User button styling. Fixes #11421

andrepereiradasilva committed:
[html5] System administrator template. Fixes #10638

jo-sf committed:
URL checking improved in com_wrapper. Fixes #4670

    When adding the scheme to an URL use correct scheme (http or https) and port.
    Checking for "http" or "https" via strstr() in the URL matched also these strings embedded in the URL (e.g. in /static/http/index.html), due to that checks changed to strpos().

zero-24 committed:
Update en-GB.com_users.ini (#11909)

wilsonge committed:
Merge branch '3.7.x' of github.com:joomla/joomla-cms into 3.7.x

mbabker committed:
Small review/cleanup in JHtml. Fixes #10771

wilsonge committed:
Merge branch 'staging' into 3.7.x

Hackwar committed:
Fix form field unit tests. Fixes #11910 Fixes #11413

andrepereiradasilva committed:
Improvements to article modals (with Create Article button). Fixes #11830

alikon committed:
[com_content] - Allow captcha when submit article. Fixes #10976

mbabker committed:
Proxy JData to the Framework. Fixes #11588

andrepereiradasilva committed:
[newsfeed modal field] Adds Create News Feed button and other improvements. Fixes #11862

alex7r committed:
Allow force refreshing the cache when getting fields in JTable. Fixes #11046

wilsonge committed:
Code review on com_content component. Fixes #10851

andrepereiradasilva committed:
[category modal field] Adds Create Category button and other improvements. Fixes #11857

Fedik committed:
Joomla.JText uses scriptOptions 'joomla.jtext'. Fixes #11690

andrepereiradasilva committed:
Remove legacy frame breaking js code. Fixes #11679

andrepereiradasilva committed:
[contact modal field] Adds Create Contact button and other improvements. Fixes #11858

mbabker committed:
Support collection of custom stats in admin stats module. Fixes #10870

brianteeman committed:
Use intro_image in newsfeed. Fixes #11402

alikon committed:
[com_tags] - Add count and link to tagged items. Fixes #10895

mbabker committed:
Allow the vote plugin's position to be configured. Fixes #11375

rdeutz committed:
Return a 403 status code instead of a 404 for a not accessible category. Fixes #11624

imanickam committed:
Add Sort by Modified Date in the Module mod_articles_news (#11128)

* Update mod_articles_news.xml

At present the sorting by date has the options to sort the articles by Published Up date and Created Date.  Adding another sort option to sort the articles by the Modified Date has been added.  This addition would be of help to list the articles that are modified recently.

* Update en-GB.mod_articles_news.ini

Added the following language string that would be used in the xml file of the module mod_articles_news.
MOD_ARTICLES_NEWS_FIELD_ORDERING_MODIFIED_DATE="Modified Date"

* Update mod_articles_news.xml

The value a.modified has been added to the Direction

dgt41 commented:
Oops something went horribly wrong here
